cmd/karma/views.go

```diff
@@ -424,6 +424,9 @@ func alerts(w http.ResponseWriter, r *http.Request) {
 					for _, am := range alert.Alertmanager {
 						for _, silence := range am.Silences {
 							if _, found := silences[am.Cluster][silence.ID]; !found {
+								if _, found :=  silences[am.Cluster]; !found {
```

Comment:
In theory it shouldn't happen. How easily can you reproduce it?
Can you print out the value of `am.Cluster` when it happens?

Reply:
we'd build and install new version to servers. bug fixed. to reproduce we need roll-back.

Reply:
I wonder if this is due to alertmanager cluster where members are sometimes getting disconnected.
If you don't see a name for your cluster via `cluster` field on each alertmanager then ClusterName() will fallback to generating a name for you based on discovery of cluster state. That name can change if some members disappear from the cluster.
If that's the case then your change is the correct fix, but that also means that lines https://github.com/prymitive/karma/blob/5e90c9df5bdbabcc41d51ab85b2d0303722c02f3/cmd/karma/views.go#L252-L257 should be deleted.
Setting `cluster` field on alertmanager instances in karma config would also prevent the crash, so you might want to try it. Although that's not really a fix, just a workaround.